docs/CppInteroperability/SwiftToCppInteroperabilityVisionDocument.md

```diff
@@ -0,0 +1,137 @@
+
+Some Swift APIs patterns will map to distinct C++ language constructs or patterns. For instance, an empty Swift enum with static members is commonly used in a namespace-like manner in Swift. This kind of enum can be mapped to a C++ namespace. It is a goal of C++ interoperability to provide a clear mapping for how Swift API patterns like this one are bridged to C++.
+
+The semantics of how Swift types behave should be preserved when they’re mapped to C++. For instance, in C++, there should still be a semantic difference between Swift value and reference types. Additionally, Swift’s copy-on-write data types like `Array` should still obey the copy-on-write semantics in C++.
+
+### Swift language evolution and API design should be unaffected
+
+Swift API authors should not change the way they write Swift code and design Swift APIs based on how specific Swift language constructs are exposed to C++. They should use the most effective Swift constructs for the task. It is a key goal of C++ interoperability that the exposed C++ interfaces are safe, performant, and ergonomic enough that programmers will not be tempted to make their Swift code worse just to make the C++ interfaces better.
+
+The design of new language features added to Swift during its evolution process should not be constrained by what can be represented in the generated C++ header. Such new features do not have to be exposed to C++ when it doesn't make sense to do so.
+
+### Objective-C support
+
+The existing Swift to Objective-C bridging layer should still be supported even when C++ bindings are generated in the generated header. Furthermore, the generated C++ bindings should use appropriate Objective-C++ types or constructs in the generated C++ declarations where it makes sense to do so. For instance, a Swift function that returns an Objective-C class instance should be usable from an Objective-C++ compilation unit.
+
+## The approach
+
+The Swift compiler exposes Swift APIs to C++ by generating a header file that contains C++ declarations that wrap around the underlying calls into Swift functions that use the native Swift calling convention. The header also provides a suitable representation for Swift types. Typically, a single header file is generated for one Swift module or module interface file.
```

Comment:
The Language Workgroup feels this would be clearer if you make the point about module interface files explicit:

```suggestion
The Swift compiler exposes Swift APIs to C++ by generating a header file that contains C++ declarations that wrap around the underlying calls into Swift functions that use the native Swift calling convention. The header also provides a suitable representation for Swift types. Typically, a single header file is generated for one Swift module. This generation can be done retroactively, starting from a Swift module interface file, and does not need to be requested as part of building the Swift module from source.
```